server/xpub-server/entities/manuscript/resolvers/uploadManuscript.js

```diff
@@ -55,18 +55,26 @@ async function uploadManuscript(_, { file, id, fileSize }, { user }) {
   }).save()
 
   const pubsub = await pubsubManager.getPubsub()
-  const reportProgress = lodash.throttle(progress => {
+
+  // Predict upload time
+  const predictedTime = 5 + 4.67e-6 * fileSize
+  const startedTime = Date.now()
+
+  const handle = setInterval(() => {
```

Comment:
Maybe to make the percentage more accurate (and account for varying upload speeds on the clients side) we could leave the call to this within the `stream.on` but do `Math.floor(progress * 50)` instead then set the interval timer once the files are uploaded to the S3 bucket? That way the first 50% progress is truly representational of the upload progress and the estimation only needs to guess ScienceBeams processing time.

Reply:
Did you read the comments on the issue?

Reply:
Let's  talk f2f... we couldn't go with this approach as most of the time is uploading to S3